### PR TITLE
Request sharing portal link from Boomi

### PR DESCRIPTION
**Pull request by itsjeyd**
cf. [OC-2505](https://tasks.opencraft.com/browse/OC-2505)

### CODE REVIEW

**Comment by bdero** [comment]
@itsjeyd Fantastic! Thanks for working on this ahead of time. I'm going to take over this PR and implement the rest since I'm assigned to [OC-2505](https://tasks.opencraft.com/browse/OC-2505).

It looks like this is the only work remaining:
* Requesting a link to the sharing portal that is associated with the learner’s Skytap environment from Boomi, passing along preferred keyboard layout, e-mail address, course name, course run.
    * **On success**: Open the link to the sharing portal that belongs to the learner’s Skytap environment in a new window (or tab).
    * **On error**: Display an appropriate error message below the form.
Collaborating with Brian Hausmann for end-to-end tests.

**Comment by bdero** [comment]
@haikuginger This PR is ready for review, although not every aspect is verifiable currently, as the live Cloudera Boomi endpoint isn't complete to the agreed specifications yet.

This PR has a good chunk of code in it, but the majority of it is for tests and grabbing settings for the XBlock. The meat and potatoes of this PR is from [this line](https://github.com/open-craft/xblock-skytap/pull/3/files#diff-d678185f8516d1615cae6845ef7d51d4R227) onwards.

At the time of writing, all of the following testing instructions should be valid and working. I tried to write them to be quick to follow and unambiguous. Let me know if you run into any issues during testing!

## Testing instructions

1. Install this version of the XBlock into a Devstack:
   ```
   pip install -e git+https://github.com/open-craft/xblock-skytap.git@request-link#egg=xblock-skytap==request-link
   ```
1. In Studio, add the Skytap XBlock to a course in studio by navigating to `Settings > Advanced Settings` and changing the "Advanced Module List" to the following:
   ```json
   [
       "skytap",
   ]
   ```
1. Make sure that `XBLOCK_SETTINGS` is currently an empty map in `lms.env.json`.
1. Create a fresh unit in the course and instantiate a new usage of the Skytap XBlock by selecting "Advanced > Skytap XBlock.
1. Now **Publish** the changes.
1. Click **View Live Version** to view the published unit in the LMS.
1. Click the **Launch** button on the Skytap XBlock's interface.
1. Observe that an error message displays *"Error: The Skytap XBlock is improperly configured."*
   Error screenshot
1. Change the `XBLOCK_SETTINGS` for `skytap` to use some fake data:
   ```json
   "XBLOCK_SETTINGS": {
       "skytap": {
           "boomi_configuration": {
               "base_url": "https://connect.boomi.example.com",
               "endpoint": "/ws/simple/createVm",
               "username": "foo",
               "token": "b8eaadd78-4ebd-4404-a261-690c8151e556"
           }
       }
   },
   ```
1. Restart the LMS and click the *Launch* button again. Observe that this time the error says "Error: An unknown error occurred while launching.". Check the LMS console and verify that a `ConnectionError` stacktrace has been logged.
1. Replace the fake `XBLOCK_SETTINGS` data with some "known working" data (see my comment on [OC-2505](https://tasks.opencraft.com/browse/OC-2505?focusedCommentId=35909&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-35909) for an example of known working data).
1. Restart the LMS and click the *Launch* button once more. Notice the button goes gray for quite a while. Be patient here, the Boomi endpoint sometimes takes over a minute to respond.
1. At the time of writing, there will be another error: *"Error: The Skytap launch service returned a malformed response."*. This happens because the Boomi endpoint is not yet finished on Cloudera's side; it currently directly returns the URL to the sharing portal, but the Skytap XBlock has been programmed to accept a JSON response containing a `SkytapURL` field. To verify that the response from Boomi was malformed, check the LMS console to make sure there was a `JSONDecodeError` stacktrace logged.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
+root = true
```
Comment (haikuginger) [inline .editorconfig:1]: Should this file be present in the repo?

File: tests/unit/test_skytap.py [hunk 2]
```diff
@@ -4,18 +4,23 @@
 
 # Imports ###########################################################
 
-import unittest
+import json
 
+import ddt
+import httpretty
 from mock import Mock
-
```
Comment (haikuginger) [inline tests/unit/test_skytap.py]: Let's keep the line break that was removed from here.

File: xblock_skytap/skytap.py [hunk 3]
```diff
@@ -129,10 +138,118 @@ def student_view(self, context):
...
+            },
+            headers={'Accept': 'application/json'},
+        )
+
+        # Handle response errors
+        try:
+            response_json = response.json()
+        except JSONDecodeError:
+            log.error('The Boomi endpoint returned the following non-JSON response content: %s', response.content)
+            self.raise_error('The Skytap launch service returned a malformed response.', exception=True)
```
Comment (haikuginger) [inline xblock_skytap/skytap.py]: In general, I think we need to tag these error messages for i18n.

**Review by haikuginger — CHANGES_REQUESTED** [review]
A couple nits, and one change request - we need to tag user-visible strings for i18n. I tested the behavior, and it looks good, so once you add i18n, then I can give a 👍.

File: xblock_skytap/skytap.py (hunk 3 above)
Comment (bdero) [reply xblock_skytap/skytap.py]: Ah, took me a second to find [an example](https://github.com/open-craft/xblock-leaderboard/blob/master/leaderboards/leaderboard.py#L34-L36), but I got this added in. Thanks!

File: tests/unit/test_skytap.py (hunk 2 above)
Comment (bdero) [reply tests/unit/test_skytap.py]: Ok, added it back in.

File: .editorconfig (hunk 1 above)
Comment (bdero) [reply .editorconfig:1]: Yup!

**Comment by bdero** [comment]
@haikuginger Thanks for the review, all comments addressed!

**Review by haikuginger — APPROVED** [review]
:+1:

- [x] I tested this: Ran through the testing script and verified that the outputs were as expected based on the given inputs.
- [x] I read through the code
- [x] I checked for accessibility issues
- [x] Includes documentation

**Pull request merged**